bottlecap/src/metrics/enhanced/metric_data.rs

```diff
@@ -0,0 +1,73 @@
+use super::constants;
+use crate::proc::proc::{self, NetworkData};
+use dogstatsd::{
+    aggregator::Aggregator,
+    metric::{Metric, MetricValue},
+};
+use std::sync::MutexGuard;
+use tracing::debug;
+use tracing::error;
+
+#[allow(clippy::module_name_repetitions)]
+pub trait EnhancedMetricData {
```

Comment:
We have a lot of struct/traits/hiearchy's in our code, I think it's increasing the already high number of LOC and serve us no favor